src/monitor.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/env python3
+# Author: Ryan Boone
+"""
+Usage: run separately from exerciser to monitor the progress/outcomes of test runs
+"""
+
+from htcondor2 import JobEventLog
+from htcondor2 import JobEventType
+import sys
+from pathlib import Path
+import os
+from datetime import datetime
+import argparse
+
+
+def parse_cla() -> argparse.Namespace:
+    """
+    Usage: command line argument parser
+    @return: parsed arguments in argparse.Namespace object
+    """
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "-v",
+        "--verbosity",
+        action="count",
+        dest="verbosity",
+        default=0,
+        help="Increases the verbosity of print stmts with each included -v"
+    )
+
+    parser.add_argument(
+        "-w",
+        "--working-dir",
+        metavar="directory_path",
+        dest="working_dir",
+        help="Specify a specific working directory for the monitor to examine.",
+    )
+
+    parser.add_argument(
+        "-t",
+        "--timestamp",
+        metavar="YYYY-MM-DD_hh-mm",
+        dest="timestamp",
+        help="Specify the timestamp of an exerciser run to target with the monitor. Must include "
+        + "entire YYYY-MM-DD_hh-mm string."
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    """
+    Usage: monitor the status of a specified exerciser run
+    """
+    args = parse_cla()
+
+    # -w option
+    # changes location of working_dir, exiting if the dir dne
+    if args.working_dir is None:
+        working_dir = Path("../working")
+    else:
+        working_dir = Path(args.working_dir)
+        if not os.path.exists(working_dir):
+            print(f"Error: Specified working directory {working_dir} does not exist")
+            sys.exit(1)
+
+    if not os.path.exists(working_dir):
+        print("Error: Couldn't find working dir. Ensure you are in src dir")
+        sys.exit(1)
+
+    if len(os.listdir(working_dir)) == 0:
+        print("Working directory is empty, nothing to monitor")
+        sys.exit(0)
+
+    # -t option
+    # specifies exerciser run to analyze. if no timestamp is provided, analyzes most recent run
+    if args.timestamp is None:
+        target_dir = None
+        for current_dir in working_dir.iterdir():
+            if target_dir is None:
+                target_dir = current_dir
+            elif target_dir.name < current_dir.name:
+                target_dir = current_dir
+    else:
+        target_dir = os.path.join(working_dir, args.timestamp)
+        if not os.path.exists(target_dir):
+            print(f"Error: Specified timestamp {args.timestamp} does not exist")
```

Comment:
```suggestion
            print(f"Error: Specified exerciser execution directory '{args.timestamp}' does not exist")
```
Consider the following... just saying `Specified timestamp` seems to vague